postgres_ffi/src/xlog_utils.rs

```diff
@@ -292,3 +292,47 @@ impl XLogRecord {
         self.xl_info == pg_constants::XLOG_SWITCH && self.xl_rmid == pg_constants::RM_XLOG_ID
     }
 }
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct XLogPageHeaderData {
+    pub xlp_magic: u16,    /* magic value for correctness checks */
+    pub xlp_info: u16,     /* flag bits, see below */
+    pub xlp_tli: u32,      /* TimeLineID of first record on page */
+    pub xlp_pageaddr: u64, /* XLOG address of this page */
+    pub xlp_rem_len: u32,  /* total len of remaining data for record */
+    padding: u32,          /* Add explicit padding */
+}
+
+impl XLogPageHeaderData {
+    pub fn from_bytes(buf: &mut Bytes) -> XLogPageHeaderData {
+        XLogPageHeaderData {
+            xlp_magic: buf.get_u16_le(),
+            xlp_info: buf.get_u16_le(),
+            xlp_tli: buf.get_u32_le(),
+            xlp_pageaddr: buf.get_u64_le(),
+            xlp_rem_len: buf.get_u32_le(),
+            padding: buf.get_u32_le(),
+        }
+    }
+}
```

Comment:
This is more flexible:
```
pub fn from_bytes(buf: &[u8])
```
You can leave the function body the same (and it still uses the `bytes::Buf` trait).

I think the `byteorder` crate also has similar tools; I don't think it matters which one we use, unless you want to be able to read directly into something that implements `std::io::Read`.

Reply:
Sorry, I forgot to post this. I didn't mean that we need to do the slice operations manually, just that the function should take `&[u8]` rather than `&mut Bytes`.